### PR TITLE
Initial database migrations

### PR DESCRIPTION
**Pull request by finnhodgkin**
Adds a books and users context with the following tables:

Books, Authors, Users, Book Requests, Book Loans, and Author/books linking table

### CODE REVIEW

File: lib/library/books/book_loan.ex [hunk 1]
```diff
@@ -0,0 +1,21 @@
+defmodule Library.Books.BookLoan do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Library.Books.BookLoan
+
+
+  schema "book_loans" do
+    field :queue, {:array, :integer}
+    field :book_id, :id
+    field :user_id, :id
```
Comment (katbow) [inline lib/library/books/book_loan.ex]: In your migration `book_id` and `user_id` are referencing other tables. Should this be 'belongs_to'  instead? Same in the `Requests` schema.

File: lib/library/books/request.ex [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    field :user_id, :id
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(%Request{} = request, attrs) do
+    request
+    |> cast(attrs, [])
+    |> validate_required([])
```
Comment (katbow) [inline lib/library/books/request.ex]: Is this necessary if nothing is required?

File: lib/library/books/book_loan.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(%BookLoan{} = book_loan, attrs) do
+    book_loan
+    |> cast(attrs, [:queue])
```
Comment (katbow) [inline lib/library/books/book_loan.ex]: Where are you casting `book_id` and `user_id`? Is this cast in an association somewhere else that I have missed?

**Comment by ZooeyMiller** [comment]
We've made these changes, but they broke all of the tests, so we're fixing those :sweat_smile:

File: lib/library/books/book.ex [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    field :isbn_13, :string
+    field :language, :string
+    field :owned, :boolean, default: false
+    field :preview_link, :string
+    field :thumbnail, :string
+    field :thumbnail_small, :string
+    field :title, :string
+    field :type, :string
+    many_to_many :author, Library.Books.Author, join_through: "author_books"
+    has_one :book_loan, Library.Books.BookLoan
```
Comment (katbow) [inline lib/library/books/book.ex]: 👍

File: lib/library/books/book_loan.ex [hunk 5]
```diff
@@ -0,0 +1,21 @@
+defmodule Library.Books.BookLoan do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Library.Books.BookLoan
+
+
+  schema "book_loans" do
+    field :queue, {:array, :integer}
+    belongs_to :user, Library.Users.User
+    belongs_to :book, Library.Books.Book
```
Comment (katbow) [inline lib/library/books/book_loan.ex]: You can also define `on_replace` and `delete` behaviour here.

**Pull request merged**